### PR TITLE
Code coverage with Istanbul/Codecov + update dev dependencies

### PR DESCRIPTION
**Pull request by danielhollas**
One more infrastructure PR to keep our tests honest and our code covered.
Several other Khan public repos are already using Codecov (e.g. wonder-blocks) so I hope it's okay to add it here as well.

I opened a test PR on my fork at  KhanovaSkola/translation-assistant#3 and the Codecov seems to work. Coverage is already great, but I'll add a couple more tests here to get closer to 100%.

I also updated dev dependencies - Mocha and Babel. For Babel, I took the same version that is in webapp and tried to use the same settings (e.g. I copied over the `.browserlistrc`).

### CODE REVIEW

**Comment by danielhollas** [comment]
@mpolyak Just wanted to get your early blessing for this PR before I finish it. :blush: 

One remaining step is to better integrate the coverage report with the babel compilation. Before I try to figure it out, I wanted to ask you whether it's a good idea to update Babel as well here, perhaps to match the same version as in webapp?

Thanks!

**Comment by danielhollas** [comment]
@mpolyak this is now ready for review/merge. I upgraded Babel and added Babel/Istanbul plugin, so the Codecov report now refers the the original source in `src/`. I verified that Manticore integration tests in the webapp still pass, tomorrow I'll test the updated code in Translation Editor locally.

**Review by mpolyak — COMMENTED** [review]
Thank you for working to improve the repo @danielhollas!

I've tried to make a production build of webapp including this branch and got an error:

```
Unexpected character '`' [./third_party/javascript-khansrc/translation-assistant/lib/math-translator.js:149,0][bb6c4311477a762ef876dcbd2f366cc4.0758231b77fa15c4f3a2.js:5413,42]
```

You can reproduce by building `webapp` with `yarn run build:webpack-prod-local`, can you please take a look?

**Comment by danielhollas** [comment]
Huh, strange. The line is 
```js
var thousandSeparatorRegex = new RegExp(`([0-9])${placeholder}([0-9])(?=[0-9]{2})`, 'g');
```
Seems perfectly fine to me, but perhaps the error is originating elsewhere?

**Comment by danielhollas** [comment]
Strange, the line is 
```js
var thousandSeparatorRegex = new RegExp(`([0-9])${placeholder}([0-9])(?=[0-9]{2})`, 'g');
```
I don't see anything wrong with it but perhaps the error is originating elsewhere.

File: lib/math-translator.js [hunk 1]
```diff
@@ -1,3 +1,33 @@
+"use strict";
+
+require("core-js/modules/es.array.concat");
```
Comment (danielhollas) [inline lib/math-translator.js:3]: @mpolyak do you think that all these new requires may be an issue?

**Comment by mpolyak** [comment]
Looks like it's an issue with UglifyJS not handling "`" characters:
```
ERROR BUILDING WEBPACK:
bb6c4311477a762ef876dcbd2f366cc4.0758231b77fa15c4f3a2.js from UglifyJs
Unexpected character '`' [./third_party/javascript-khansrc/translation-assistant/lib/math-translator.js:149,0][bb6c4311477a762ef876dcbd2f366cc4.0758231b77fa15c4f3a2.js:5413,42]
```
If I look in the before/after of the build files in this PR, before it was producing concated strings "..." + " ...", whereas now it (webpack) leaves them as template literals, which apparently makes uglifyjs choke :/

**Comment by mpolyak** [comment]
Maybe try https://babeljs.io/docs/en/babel-plugin-transform-template-literals. Although it doesn't look like that plugin is used in webapp, which is confusing as to why it's not an issue in other code.

**Comment by mpolyak** [comment]
Maybe something in `webapp/babel.config.js` auto includes `@babel/plugin-transform-template-literals` as I do see it installed in `/node_modules`.

**Comment by mpolyak** [comment]
I got some further information that inclusion of`plugin-transform-template-literals` is part babel `preset-env`, and the `.browserslistrc` list. There must be some config difference between this PR and how webapp does it.

**Comment by mpolyak** [comment]
Changing `.babelrc.json` to:
```
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "useBuiltIns": "usage",
        "loose": true,
        "corejs": {
          "version": 3
        }
      }
    ]
  ],
  "plugins": [
    [
      "@babel/plugin-transform-template-literals",
      {
        "loose": true
      }
    ]
  ]
}
```

Seems to resolve the issue. I wonder why the `"@babel/preset-env"` preset is not doing it. It looks similar to the webapp `babel.config.js`:

```
const presets = [
    [
        /**
         * Uses .browserslist to determine which plugins and polyfills
         * to use.
         * https://babeljs.io/docs/en/babel-preset-env
         */
        "@babel/env",
        {
            /**
             * Adds specific imports for polyfills when they are
             * used in each file. We take advantage of the fact
             * that a bundler will load the same polyfill only once.
             * https://babeljs.io/docs/en/babel-preset-env#usebuiltins
             */
            useBuiltIns: "usage",

            /**
             * This must match the version of core-js in package.json
             * or babel will complain.
             */
            corejs: {
                version: 3,
            },
        },
    ],

    /**
     * Compile JSX to React.createElement()
     * https://babeljs.io/docs/en/babel-preset-react
     */
    "@babel/react",
];
```

**Comment by danielhollas** [comment]
@mpolyak thanks for getting to the bottom of this. The babel doc mentions UglifyJS specifically as being only able to understand ES5. See https://babeljs.io/docs/en/babel-preset-env#forcealltransforms

I wonder whether we should tweak the config further according to the docs above so that we prevent similar hiccups in the future?

Why the webapp build works with its current config is indeed strange.

**Comment by danielhollas** [comment]
FYI I tested this branch locally and ST seem to work in the Translation Editor. So hopefully this is good to go.

**Comment by mpolyak** [comment]
@danielhollas I did more pocking around and I think it's an issue with the contents of `.browserlistrc`. I changed it to:

```
# Browsers that we support

[modern]
Chrome 41 # used by googlebot
last 1 Edge version
last 1 Firefox version
last 1 Safari version # desktop Safari
last 1 iOS version # mobile Safari

[ssr]
node 8
```

Removed the "plugins" section from the babel config, and now it transpiles the template strings!

Can you give that a try please?

**Comment by danielhollas** [comment]
@mpolyak thanks! Just tested everything locally and this does the trick! I also looked at the differences at the compiled file in this commit and it is now mostly white-space changes so :crossed_fingers:

**Review by mpolyak — APPROVED** [review]
Builds locally for prod and as ZND: https://prod-znd-201021-michaelp-5.khanacademy.org/, thanks @danielhollas!

**Pull request merged**